### PR TITLE
add missing lints in analysis_options.yaml

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

**Comment by pq** [comment]
Thinking out loud: should we consider adding a validation step to the tests to ensure this stays up to date?  See also #1051, which has I think some relevant conversation.

File: analysis_options.yaml [hunk 1]
```diff
@@ -16,20 +16,30 @@ linter:
     - annotate_overrides
     - avoid_annotating_with_dynamic
 #    - avoid_as
+#    - avoid_bool_literals_in_conditional_expressions # not yet tested
```
Comment (pq) [inline analysis_options.yaml]: Actually, on reading this again, I'm not sure this note is totally accurate.  Maybe a link to #1068 instead?  Maybe something like:

    # under review (see #1068)

**Comment by a14n** [comment]
PTAL

**Comment by pq** [comment]
Super.  Thanks!

**Pull request merged**